src/classes/AuthenticationSession.ts

```diff
@@ -178,4 +181,18 @@ export class AuthenticationSession {
       this.authContext.reject();
     });
   }
+
+  private obtainNewAuthTokenUsingRefreshToken(): void {
+    this.fusionAuthClient.exchangeRefreshTokenForAccessToken(this.refreshToken, '', '', '', '')
+      .then((clientResponse) => {
+        this.authToken = clientResponse.response.access_token ?? '';
+      })
```

Comment:
You probably need to also update the refresh token here - ideally, the response should include both a new access token and a new refresh token, so that the authentication server can defend against token replay attacks. I don't know if FusionAuth specifically does that, but if it gives you a refresh token in the response - even if in practice it's the same - you should update your copy.

Reply:
Thanks for the input Jason.

Currently, I can't get Fusionauth to include refresh tokens in its responses. (Despite the checkbox being checked in the admin, not sure if you know anything about that too)

Reply:
You need to include the `offline_access` scope in your initial request: https://fusionauth.io/docs/v1/tech/oauth/tokens#refresh-token . It's been a bit since I've seen their admin page, but I'd expect that checking that box *allows* the selected client to request refresh tokens, rather than configuring FusionAuth to give them out unasked-for.